src/EA.Weee.DataAccess.Tests.DataAccess/StoredProcedure/SpgProducerEeeCsvDataTests.cs

```diff
@@ -65,101 +65,10 @@ public async Task Execute_HappyPath_ReturnsProducerEeeWithSelectedYearAndObligat
-                }
-
-                helper.CreateEeeOutputAmount(dataReturnVersion1, producer2.RegisteredProducer, b2bObligationType, 2, 400);
-
-                db.Model.SaveChanges();
-
-                // Act
-                var results = await db.StoredProcedures.SpgProducerEeeCsvData(complianceYear, null, b2cObligationType, false, false);
-
-                //Assert
-                Assert.NotNull(results);
-
-                //Data return with obligation type B2B should not be there in the result.
-                ProducerEeeCsvData b2bProducer = results.Find(x => (x.PRN == pnr2));
-                Assert.Null(b2bProducer);
-
-                ProducerEeeCsvData result = results[0];
-                Assert.Equal(approvalNumber, result.ApprovalNumber);
-                Assert.Equal(pnr1, result.PRN);
-                Assert.Equal(10, result.Cat1Q1);
-                Assert.Equal(20, result.Cat2Q1);
-                Assert.Equal(30, result.Cat3Q1);
-                Assert.Equal(40, result.Cat4Q1);
-                Assert.Equal(50, result.Cat5Q1);
-                Assert.Equal(60, result.Cat6Q1);
-                Assert.Equal(70, result.Cat7Q1);
-                Assert.Equal(80, result.Cat8Q1);
-                Assert.Equal(90, result.Cat9Q1);
-                Assert.Equal(100, result.Cat10Q1);
-                Assert.Equal(110, result.Cat11Q1);
-                Assert.Equal(120, result.Cat12Q1);
-                Assert.Equal(130, result.Cat13Q1);
-                Assert.Equal(140, result.Cat14Q1);
-                Assert.Equal(150, result.Cat15Q1);
-
-                Assert.Equal(5, result.Cat1Q2);
-                Assert.Equal(10, result.Cat2Q2);
-                Assert.Equal(15, result.Cat3Q2);
-                Assert.Equal(20, result.Cat4Q2);
-                Assert.Equal(25, result.Cat5Q2);
-                Assert.Equal(30, result.Cat6Q2);
-                Assert.Equal(35, result.Cat7Q2);
-                Assert.Equal(40, result.Cat8Q2);
-                Assert.Equal(45, result.Cat9Q2);
-                Assert.Equal(50, result.Cat10Q2);
-                Assert.Equal(55, result.Cat11Q2);
-                Assert.Equal(60, result.Cat12Q2);
-                Assert.Equal(65, result.Cat13Q2);
-                Assert.Equal(70, result.Cat14Q2);
-                Assert.Equal(75, result.Cat15Q2);
-
-                Assert.Null(result.Cat1Q3);
-                Assert.Null(result.Cat1Q4);
-
-                Assert.Equal(1800, result.TotalTonnage);
-            }
-        }
-
-        [Fact]
-        public async Task Execute_HappyPath_ReturnsProducerEeeWithSelectedYearAndObligationType_As_B2B_WithAllCategories()
+        [Theory]
+        [InlineData(ObligationType.B2B)]
+        [InlineData(ObligationType.B2C)]
+        public async Task Execute_HappyPath_ReturnsProducerEeeWithSelectedYearAndObligationType_As_WithAllCategories(ObligationType obligationType)
```

Comment:
I kind of like what you've done here (it wasn't what I had in mind!) -- although I did suggest you do Both as well -- but I think the test would be better if it did not have the various conditions in that are making assumptions about the parameter (it's assuming in the else that Both wasn't provided, but I think it would be a useful test to supply it). Maybe always make producer1 use the type specified, but you might need an additional parameter to specify what to expect. I'm not quite sure yet how it would work, but my thinking is the InlineData specification should encode the expected logic, so the test isn't making decisions about what to call and what checks to do